src/gui/inputdevices/KeyboardInputHandler.cpp

```diff
@@ -13,10 +13,10 @@ KeyboardInputHandler::KeyboardInputHandler(InputContext* inputContext): Abstract
 KeyboardInputHandler::~KeyboardInputHandler() = default;
 
 auto KeyboardInputHandler::handleImpl(InputEvent* event) -> bool {
-    auto keyEvent = reinterpret_cast<GdkEventKey*>(event->sourceEvent);
     GtkXournal* xournal = inputContext->getXournal();
 
-    if (keyEvent->type == GDK_KEY_PRESS) {
+    if (event->sourceEvent->type == GDK_KEY_PRESS) {
+        auto keyEvent = reinterpret_cast<GdkEventKey*>(event->sourceEvent);
```

Comment:
Could you remove the indirection for ever call and store the source event in a variable? It would prevent future accidential uses of the wrong event.

Reply:
Not entirely sure what you meant, but I changed the code to use `GdkEvent* gdkEvent = event->sourceEvent;` and then replaced every `event->sourceEvent` with `gdkEvent`.

Reply:
That was exactly what I meant.